packages/common-utils/src/metadata.ts

```diff
@@ -254,13 +268,24 @@ export class Metadata {
       }}) as keysArr
       FROM ${tableExpr({ database: databaseName, table: tableName })} ${where}`;
     } else {
-      sql = chSql`SELECT DISTINCT lowCardinalityKeys(arrayJoin(${{
-        Identifier: column,
-      }}.keys)) as key
-      FROM ${tableExpr({ database: databaseName, table: tableName })} ${where}
-      LIMIT ${{
-        Int32: maxKeys,
-      }}`;
+      sql = chSql`
+        WITH sampledKeys as (
+          SELECT ${{
+            Identifier: column,
+          }}.keys AS keysArr
+          FROM ${tableExpr({ database: databaseName, table: tableName })} ${where}
+          LIMIT ${{
```

Comment:
I'm concerned this method will result in high memory usage or poor performance since we might be storing all the column values first (all millions of values) and then doing a distinct on it later (I'm not sure if this gets optimized away by ClickHouse).